### ISSUE TITLE
Crash on windows qemu guest with vfio-scsi (child_length not even multiple)

### ISSUE THREAD

**Issue by gnif** [spdk/spdk#977]
When using a RAID0 bdev that spans two NVMe drives (Samsung 960 Pro 1TB) as a virtio scsi lun for Qemu some accesses will cause the vhost to hang with `child_length 57952 not even multiple of lba_size 512`

## Expected Behavior
No crash, no error, etc.

## Current Behavior

The guest VM's IO hangs and the vhost logs the following.
```
nvme_ns_cmd.c: 252:_nvme_ns_cmd_split_request_prp: *ERROR*: child_length 57952 not even multiple of lba_size 512
nvme_ns_cmd.c: 252:_nvme_ns_cmd_split_request_prp: *ERROR*: child_length 57952 not even multiple of lba_size 512
```
## Steps to Reproduce

1. `./scripts/setup.sh`
2. `./app/vhost/vhost -S /var/tmp -m 0x3`
3. `./scripts/rpc.py bdev_nvme_attach_controller -b NVMe1 -t PCIe -a 0000:08:00.0`
4. `./scripts/rpc.py bdev_nvme_attach_controller -b NVMe2 -t PCIe -a 0000:41:00.0`
5. `./scripts/rpc.py bdev_raid_create -n WinRaid0 -z 64 -r 0 -b "NVMe1n1 NVMe2n1"`
6. `./scripts/rpc.py construct_vhost_scsi_controller vhost.0`
7. `./scripts/rpc.py add_vhost_scsi_lun vhost.0 0 WinRaid0`
8. Launch qemu with `-chardev socket,id=char0,path=/var/tmp/vhost.0 -device  vhost-user-scsi-pci,id=scsi0,chardev=char0`
9. Install Windows 10
10. Obtain and run UserBenchmark from https://www.userbenchmark.com/

It seems it isn't always reproducible, a repeat test of this configuration ran just fine.

## Context (Environment including OS version, SPDK version, etc.)

CPU: AMD Threadripper 1950X in NUMA mode
Host: Debian 10
SPDK: Git master @ 23975858e5c1b21fd5328e4e05c1fcaf87a65ebf

**Comment by gnif** [comment]
I have been digging through this solo since the Slack channel link is invalid and I don't seem to be able to get any help. I have narrowed this down to the raid bdev as it's reproducible on malloc bdevs.

The issue can be reliably triggered by running HD Tune Pro's 'Random seek / size 8MB' test on the 'Extra tests' tab. This test will 100% of the time trigger the device to hang. Adding additional logging to SPDK it's become clear that the raid bdev is reading too much data, I am still to figure out what is causing this.

```
vhost_blk.c: 436:process_vq: *DEBUG*: ====== Starting processing request idx 73======
vhost_blk.c: 308:process_blk_request: *DEBUG*: read block:0x16afe08c200 len:179200.
spdk_bdev_readv: offset:0xb57f0461 num_blocks:0x0000015e.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe08c R idx:0 off:0x00000001 lba:0x5abf8231 blocks:7.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe08d R idx:1 off:0x00000000 lba:0x5abf8230 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe08e R idx:0 off:0x00000000 lba:0x5abf8238 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe08f R idx:1 off:0x00000000 lba:0x5abf8238 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe090 R idx:0 off:0x00000000 lba:0x5abf8240 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe091 R idx:1 off:0x00000000 lba:0x5abf8240 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe092 R idx:0 off:0x00000000 lba:0x5abf8248 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe093 R idx:1 off:0x00000000 lba:0x5abf8248 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe094 R idx:0 off:0x00000000 lba:0x5abf8250 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe095 R idx:1 off:0x00000000 lba:0x5abf8250 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe096 R idx:0 off:0x00000000 lba:0x5abf8258 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe097 R idx:1 off:0x00000000 lba:0x5abf8258 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe098 R idx:0 off:0x00000000 lba:0x5abf8260 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe099 R idx:1 off:0x00000000 lba:0x5abf8260 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe09a R idx:0 off:0x00000000 lba:0x5abf8268 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afe09b R idx:1 off:0x00000000 lba:0x5abf8268 blocks:8.
vhost_blk.c: 463:process_vq: *DEBUG*: ====== Task 0x20216b55def0 req_idx 73 submitted ======
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdd9e R idx:0 off:0x00000000 lba:0x5abf7678 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdd9f R idx:1 off:0x00000000 lba:0x5abf7678 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda0 R idx:0 off:0x00000000 lba:0x5abf7680 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda1 R idx:1 off:0x00000000 lba:0x5abf7680 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda2 R idx:0 off:0x00000000 lba:0x5abf7688 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda3 R idx:1 off:0x00000000 lba:0x5abf7688 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda4 R idx:0 off:0x00000000 lba:0x5abf7690 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda5 R idx:1 off:0x00000000 lba:0x5abf7690 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda6 R idx:0 off:0x00000000 lba:0x5abf7698 blocks:8.
bdev_raid.c: 323:raid0_submit_rw_request: *DEBUG*: start_strip:0x16afdda7 R idx:1 off:0x00000000 lba:0x5abf7698 blocks:8.
```

Here you can see that the read doesn't stop even though the task has completed.

**Comment by gnif** [comment]
This may be related to #979

**Comment by gnif** [comment]
It seems the fix in #979 has corrected this, however, the trigger for it was random so I can not be certain it was the same issue. I will close this for now and if I see it again I will re-open.
